### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by m-cat**
- Remove deprecated 'CustomUserIDOptions'
- Fix path functions, convert domain to lowercase when sanitizing

### CODE REVIEW

**Review by ro-tex — APPROVED** [review]

File: src/paths.test.ts [hunk 1]
```diff
@@ -16,24 +19,35 @@ describe("parentPath", () => {
   const paths: Array<[string, string | null]> = [
     ["app.hns/path/file.json", "app.hns/path"],
     ["app.hns///path///file.json", "app.hns/path"],
+    ["APP.hns///Path///file.json", "app.hns/Path"],
     ["app.hns//path", "app.hns"],
     ["app.hns/path/", "app.hns"],
+    ["/app.hns/path/", "/app.hns"],
```
Comment (kwypchlo) [inline src/paths.test.ts:25]: this should be null too, I think ?

Comment (dghelm) [reply src/paths.test.ts:25]: What "dataDomain" is this in? Is this a relative path, a null dataDomain or do we just ignore the leading '/'?
if 1, I don't know how to get an absolute path
2, How does an app get permissions to a null dataDomain?
3, I think it's fine to accept someone does this as a strange convention, but would it be better to just throw an error in development so they conform to path-style we expect?

Comment (m-cat) [reply src/paths.test.ts:25]: I am thinking we disallow the leading / for now. We can always allow it later if there's a use case. Allowing it now and then removing it later would be harder.

File: src/paths.ts [hunk 2]
```diff
@@ -6,8 +6,19 @@ import { removeAdjacentChars, trimSuffix } from "./utils";
...
-  return path.split("/")[0];
+export function getPathDomain(path: string): string | null {
+  // Sanitize the path.
+  const sanitizedPath = sanitizePath(path);
+  if (sanitizedPath === null) {
+    return null;
+  }
+
+  // Extract the domain.
+  const domain = sanitizedPath.split("/")[0];
```
Comment (kwypchlo) [inline src/paths.ts:17]:
```suggestion
  const [domain] = sanitizedPath.split("/");
```

**Review by kwypchlo — APPROVED** [review]

**Pull request merged**